sdk/core/azure-core-amqp/src/amqp/link.cpp

```diff
@@ -303,6 +303,29 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
     }
   }
 
+  void LinkImpl::SubscribeToDetachEvent(OnLinkDetachEvent onLinkDetach)
+  {
+    m_onLinkDetachEvent = std::move(onLinkDetach);
+    m_linkSubscriptionHandle
+        = link_subscribe_on_link_detach_received(m_link, OnLinkDetachEventFn, this);
+  }
+
+  void LinkImpl::UnsubscribeFromDetachEvent()
+  {
+    if (m_linkSubscriptionHandle != nullptr)
+    {
+      link_unsubscribe_on_link_detach_received(m_linkSubscriptionHandle);
+    }
+  }
+  void LinkImpl::OnLinkDetachEventFn(void* context, ERROR_HANDLE error)
```

Comment:
Is this void* vs typed because the callers don't know what type to send us?  Can you add a comment into code on why this must be a void*.

Reply:
`OnLinkDetachEventFn` (and every other `Fn` function) implements a function prototype defined in C. By convention the first parameter is a context parameter, but only the code which defines the function callback knows what that type is. Thus it is `void *`.